circle.yml

```diff
@@ -1,6 +1,83 @@
+# These environment variables must be set in CircleCI UI
+#
+# DOCKERHUB_REPO - docker hub repo, format: <username>/<repo>
+# DOCKER_EMAIL   - login info for docker hub
+# DOCKER_USER
+# DOCKER_PASS
+
 machine:
+  pre:
+    - curl -sSL https://s3.amazonaws.com/circle-downloads/install-circleci-docker.sh | bash -s -- 1.10.0
   node:
     version: 4.3.1
+  services:
+    - docker
+
 dependencies:
+  cache_directories:
+    - "~/docker"
   pre:
     - npm install -g npm@3
+    - sudo apt-get update; sudo apt-get install pigz
+    - sudo pip install tox mozdownload mozinstall
+    - mozdownload --version latest --destination firefox.tar.bz2
+    - mozinstall firefox.tar.bz2
+    - wget -O geckodriver.tar.gz https://github.com/mozilla/geckodriver/releases/download/v0.10.0/geckodriver-v0.10.0-linux64.tar.gz
```

Comment:
If Circle meets all our requirements then running all tests there probably makes more sense since we ideally want to say that the containers built from a current revision pass all the relevant tests. 

Where it gets complicated is that addons-frontend is not just one app, the unittests cover all apps (disco (discovery pane), amo (mobile pages) and the admin app) but the current uitests are only specific to just one app at the moment although this will likely change.